packages/kbn-ftr-common-functional-services/services/retry/retry_for_success.ts

```diff
@@ -44,27 +44,45 @@ interface Options<T> {
   onFailureBlock?: () => Promise<T>;
   onFailure?: ReturnType<typeof defaultOnFailure>;
   accept?: (v: T) => boolean;
+  description?: string;
   retryDelay?: number;
+  retryCount?: number;
 }
 
 export async function retryForSuccess<T>(log: ToolingLog, options: Options<T>) {
```

Comment:
Have you investigated if `p-retry` is applicable here? Additionally exponential back off timeout may work better.

Reply:
@dmlemeshko Could you share found issues with `p-retry`?

My observation is `p-retry` feels like a quite popular library with 14M weekly downloads. A lot of bugs should've been fixed already. On top of that `p-retry` provides flexible configuration like exponential back off retry timeout while it still can be configured linearly. It allows to abort retries as well. Checking Kibana's codebase it's not hard to see it's used in multiple packages and plugins.

Having a comment for `retryForSuccess` with some explanation and improvements over `p-retry` will be really helpful for future maintenance.

Reply:
@maximpn Here is the [ticket](https://github.com/elastic/kibana/issues/178535) that I explained what challenge we faced, and in our case, we didn't have any log about the retry attempts which made it hard for us to understand if it was one request that timeout or there were multiple requests and due to exponential back off, the overall attempt failed. (The way that we tested it locally was by throwing an error in the test function and setting the retry to 10 times)

In general, I think it would be better to only rely on one library/package for retry purposes, for easier maintenance and a better understanding of the tool.

Reply:
@maryam-saeidi thanks for the details 👍

> we didn't have any log about the retry attempts
Yes, it sounds like a bug. Or exponential timeout was too high so the test was interrupted due to a timeout. Btw, you could configure `p-retry` to have a constant retry interval.